include/deal.II/base/parameter_handler.h

```diff
@@ -1597,6 +1625,19 @@ class ParameterHandler : public Subscriptor
   virtual bool read_input (const std::string &filename,
                            const bool optional = false,
                            const bool write_stripped_file = false);
+  /**
+   * This is a shell function of read_input(std::string, bool, bool) with a
+   * compact argument list.
+   *
+   * This function will set arguments and flags accroding to the provided
+   * ReadInputFlags and call read_input(std::string, bool, bool). The
+   * ReadInputFlags passed to this function only has effect on this time of
+   * reading.
+   *
+   * Return what read_input(std::string, bool, bool) returned.
```

Comment:
If you really want to deprecate the old read_input function, I will have to make implementation of the new read_input to be independent of the old read_input function also, not only the documentation.